### PR TITLE
Adding a wait for federation apiserver to be ready in e2e tests

### PR DESCRIPTION
**Pull request by nikhiljindal**
Ref #26762

@kubernetes/sig-cluster-federation @mml

### CODE REVIEW

**Comment by nikhiljindal** [comment]
Am testing this out

File: test/e2e/framework/util.go [hunk 1]
```diff
@@ -836,6 +836,18 @@ func WaitForDefaultServiceAccountInNamespace(c *client.Client, namespace string)
 	return waitForServiceAccountInNamespace(c, namespace, "default", ServiceAccountProvisionTimeout)
 }
 
+// WaitForFederationApiserverReady waits for the federation apiserver to be ready.
+// It tests the readiness by sending a GET request and expecting a non error response.
+func WaitForFederationApiserverReady(c *federation_internalclientset.Clientset) error {
+	return wait.PollImmediate(time.Second, 1*time.Minute, func() (bool, error) {
```
Comment (quinton-hoole) [inline test/e2e/framework/util.go]: Please use one of the standard timeouts in utils. 

https://github.com/kubernetes/kubernetes/blob/master/docs/devel/writing-good-e2e-tests.md

Perhaps ServiceRespondingTimeout?

https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/util.go#L131

**Comment by quinton-hoole** [comment]
@nikhiljindal As I mentioned, I'm pretty sure that this is not the cause of the errors we're seeing (e.g. my Federation API Server has been running for over an hour, and is healthy, but I still see that panic), but it seems like a reasonable change to make the e2e tests more robust.  Lets hold off merging this until we've found the root cause though.  @mml mentioned that he'd tacked this down to one of the other AfterEach clauses setting the client to nil, leading to this nil pointer panic.

```
•! Panic in Spec Teardown (AfterEach) [42.099 seconds]
[k8s.io] Federation apiserver [Feature:Federation]
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:660
  should allow creation and deletion of cluster api objects [AfterEach]
  /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/federation-apiserver.go:103

  Test Panicked
  runtime error: invalid memory address or nil pointer dereference
  /usr/local/go/src/runtime/panic.go:443

  Full Stack Trace
  	/usr/local/go/src/runtime/panic.go:443 +0x4e9
  k8s.io/kubernetes/pkg/client/unversioned.(*namespaces).Get(0xc8200d0530, 0x20484e0, 0xa, 0x0, 0x0, 0x0)
  	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/client/unversioned/namespaces.go:100 +0x73
  k8s.io/kubernetes/test/e2e/framework.SkipUnlessFederated(0x0)
  	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/util.go:395 +0x10d
  k8s.io/kubernetes/test/e2e.glob.func31.1()
  	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/federation-apiserver.go:38 +0x3b
  k8s.io/kubernetes/vendor/github.com/onsi/ginkgo/internal/leafnodes.(*runner).runSync(0xc8200cbaa0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
  	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/e2e.go:239 +0x3c5
  k8s.io/kubernetes/test/e2e.TestE2E(0xc8200a3950)
  	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/e2e_test.go:30 +0x21
  testing.tRunner(0xc8200a3950, 0x394e3b0)
  	/usr/local/go/src/testing/testing.go:473 +0x98
  created by testing.RunTests
  	/usr/local/go/src/testing/testing.go:582 +0x892

```

**Comment by nikhiljindal** [comment]
Yes, This PR doesnt fix the panic.
This fixes the issue where create cluster requests are failing.

**Comment by quinton-hoole** [comment]
@nikhiljindal Ack.  Lets get this merged.

**Comment by nikhiljindal** [comment]
rebased

**Pull request merged**